### ISSUE TITLE
Feature request: Add a display option to hide header and footer

### ISSUE THREAD

**Issue by mskar** [htop-dev/htop#168]
Please add an option to hide the header and footer.

What I see:

![image](https://user-images.githubusercontent.com/13444106/93837245-5e221b00-fc53-11ea-9867-5f27b6aa8dbe.png)

What I want to see (header and footer hidden):
![image](https://user-images.githubusercontent.com/13444106/93837278-772acc00-fc53-11ea-9646-603b02278f6e.png)

**Comment by fasterit** [comment]
Remove the meters from the left and right columns and disable "Leave a margin around header".
You'll have the top empty as you want it and just have to live with the footer for now.

**Comment by BenBE** [comment]
If I see it correctly we could go about this by not adding the header and footer to the set of blocks to render in the main display routine. I'll take a look at this if I can build a patch for this. Shouldn't be too hard.

A quick&dirty change to get rid of the header is this patch:
```diff
diff --git a/Header.c b/Header.c
index d2f3b88..3e3b6a1 100644
--- a/Header.c
+++ b/Header.c
@@ -156,6 +156,8 @@ void Header_reinit(Header* this) {
 }
 
 void Header_draw(const Header* this) {
+   return;
+
    int height = this->height;
    int pad = this->pad;
    attrset(CRT_colors[RESET_COLOR]);
@@ -177,6 +179,8 @@ void Header_draw(const Header* this) {
 }
 
 int Header_calculateHeight(Header* this) {
+   return 0;
+
    int pad = this->settings->headerMargin ? 2 : 0;
    int maxHeight = pad;
 
```

This doesn't handle the case of the function bar yet, as the handling of the function bar is a bit more distributed across the code base.

While looking at the code base I noticed that vertical alignment of panels is not yet implemented; a change that would make implementation of this change a bit easier as this really would just be hiding the function bar under some circumstances. Unfortunately hiding the function bar makes the overall screen layout sensitive to keyboard input as entering incremental search mode would have to display the function bar and thus cause a resize of all panels to be triggered.

**Comment by fasterit** [comment]
I think we can just leave the footer as is. No need to be able to configure everything and add that much complexity as a result.

**Comment by BenBE** [comment]
Which leaves deciding on a keyboard shortcut for it (no need for CLI option nor permanent setting IMHO).

**Comment by fasterit** [comment]
not either
esp. as we disable / enable UI elements via settings and not via keyboard shortcuts
consistency++

**Comment by BenBE** [comment]
I think a "quick switch" for the header would be quite useful. removing and reconfiguring the header as I use it ATM would be quite a hassle each time.

**Comment by fasterit** [comment]
Yes, that may make sense for the many-core users where there's a good piece of space taken up by the meters at the top.
The configuration switch could then be "Hide header by default" that could be enabled in settings, so a key has to be pressed to show it for those that prefer it this way around.

**Comment by kevinhwang91** [comment]
I don't like the footer which is redundant for me, and UI is different from `top`.

**Comment by cgzones** [comment]
The function bar can be hidden since a3cced9fb615a1cfe0b334fc41dc785f1fb9aef5 and hiding the header will be easy with #445.

**Comment by kevinhwang91** [comment]
> The function bar can be hidden since [a3cced9](https://github.com/htop-dev/htop/commit/a3cced9fb615a1cfe0b334fc41dc785f1fb9aef5) and hiding the header will be easy with #445.

It works for me, thanks.
